dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/io/readers/SubSnpCoreFieldsReader.java

```diff
@@ -53,139 +53,79 @@
     }
 
     @Override
-    public SubSnpCoreFields read() throws Exception {
+    protected void openCursor(Connection connection) {
         try {
-            return super.read();
-        } catch (BadSqlGrammarException e) {
-            throw new SQLException("Build " + dbsnpBuild + " does not exist", e);
+            connection.setAutoCommit(false);
+        } catch (SQLException e) {
+            throw new RuntimeException("Failed to set autocommit=false", e);
         }
+        super.openCursor(connection);
     }
 
-    private String buildSql(int dbsnpBuild) throws Exception {
+    private String buildSql(String assembly) throws Exception {
         String sql =
-                "SELECT distinct " +
-                        "sub.subsnp_id AS " + SUBSNP_ID_COLUMN +
-                        ",loc.snp_id AS " + REFSNP_ID_COLUMN +
-                        ",hgvs.hgvs_c AS " + HGVS_C_STRING +
-                        ",hgvs.start_c+1 AS " + HGVS_C_START +
-                        ",hgvs.stop_c+1 AS " + HGVS_C_STOP +
-                        ",hgvs.ref_allele_c AS " + REFERENCE_C +
-                        ",hgvs.hgvs_t AS " + HGVS_T_STRING +
-                        ",hgvs.start_t+1 AS " + HGVS_T_START +
-                        ",hgvs.stop_t+1 AS " + HGVS_T_STOP +
-                        ",hgvs.ref_allele_t AS " + REFERENCE_T +
-                        ",hgvs.var_allele AS " + ALTERNATE +
-                        ",obsvariation.pattern AS " + ALLELES +
-                        ",ctg.contig_name AS " + CONTIG_NAME_COLUMN +
-                        ",loc.asn_from +1 AS " + CONTIG_START_COLUMN +
-                        ",loc.asn_to +1 AS " + CONTIG_END_COLUMN +
-                        ",loc.loc_type AS " + LOC_TYPE_COLUMN +
-                        ",ctg.contig_chr AS " + CHROMOSOME_COLUMN +
-                        ",loc.phys_pos_from + 1 AS " + CHROMOSOME_START_COLUMN +
-                        ",loc.phys_pos_from + 1 + loc.asn_to - loc.asn_from AS " + CHROMOSOME_END_COLUMN +
-                        ",batch.loc_batch_id_upp AS " + BATCH_COLUMN +
-                        ",CASE " +
-                        "   WHEN hgvs.orient_c = 2 THEN -1 ELSE 1 " +
-                        "END AS " + HGVS_C_ORIENTATION +
-                        ",CASE " +
-                        "   WHEN hgvs.orient_t = 2 THEN -1 ELSE 1 " +
-                        "END AS " + HGVS_T_ORIENTATION +
-                        ",CASE " +
-                        "   WHEN loc.orientation = 1 THEN -1 ELSE 1 " +
-                        "END AS " + SNP_ORIENTATION_COLUMN +
-                        ",CASE " +
-                        "   WHEN ctg.orient = 1 THEN -1 ELSE 1 " +
-                        "END AS " + CONTIG_ORIENTATION_COLUMN +
-                        ",CASE " +
-                        "   WHEN link.substrand_reversed_flag = 1 THEN -1 ELSE 1 " +
-                        "END AS " + SUBSNP_ORIENTATION_COLUMN +
+                "SELECT " +
```

Comment:
I don't know about others, but given the volume of data we are dealing with, I agree it is good to have a reader that excludes the genotypes and allele frequencies, and another that reads everything.